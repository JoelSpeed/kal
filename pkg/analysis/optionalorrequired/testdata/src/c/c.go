package c

type OptionalOrRequiredTestStruct struct {
	RequiredEnumField RequiredEnum // want "field RequiredEnumField must be marked as optional or required"

	KubebuilderRequiredEnumField KubeBuilderRequiredEnum // want "field KubebuilderRequiredEnumField must be marked as optional or required"

	OptionalEnumField OptionalEnum // want "field OptionalEnumField must be marked as optional or required"

	KubebuilderOptionalEnumField KubeBuilderOptionalEnum // want "field KubebuilderOptionalEnumField must be marked as optional or required"
}

// +required
// +kubebuilder:validation:Enum=Foo;Bar;Baz
type RequiredEnum string // want "type RequiredEnum should not be marked as required"

// +kubebuilder:validation:Required
// +kubebuilder:validation:Enum=Foo;Bar;Baz
type KubeBuilderRequiredEnum string // want "type KubeBuilderRequiredEnum should not be marked as kubebuilder:validation:Required"

// +optional
// +kubebuilder:validation:Enum=Foo;Bar;Baz
type OptionalEnum string // want "type OptionalEnum should not be marked as optional"

// +kubebuilder:validation:Optional
// +kubebuilder:validation:Enum=Foo;Bar;Baz
type KubeBuilderOptionalEnum string // want "type KubeBuilderOptionalEnum should not be marked as kubebuilder:validation:Optional"
