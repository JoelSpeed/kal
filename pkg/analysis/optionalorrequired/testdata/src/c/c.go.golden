package c

type OptionalOrRequiredTestStruct struct {
	RequiredEnumField RequiredEnum // want "field RequiredEnumField must be marked as kubebuilder:validation:Optional or kubebuilder:validation:Required"

	KubebuilderRequiredEnumField KubeBuilderRequiredEnum // want "field KubebuilderRequiredEnumField must be marked as kubebuilder:validation:Optional or kubebuilder:validation:Required"

	OptionalEnumField OptionalEnum // want "field OptionalEnumField must be marked as kubebuilder:validation:Optional or kubebuilder:validation:Required"

	KubebuilderOptionalEnumField KubeBuilderOptionalEnum // want "field KubebuilderOptionalEnumField must be marked as kubebuilder:validation:Optional or kubebuilder:validation:Required"
}

// +kubebuilder:validation:Enum=Foo;Bar;Baz
type RequiredEnum string // want "required should not be defined on a type RequiredEnum"

// +kubebuilder:validation:Enum=Foo;Bar;Baz
type KubeBuilderRequiredEnum string // want "kubebuilder:validation:Required should not be defined on a type KubeBuilderRequiredEnum"

// +kubebuilder:validation:Enum=Foo;Bar;Baz
type OptionalEnum string // want "optional should not be defined on a type OptionalEnum"

// +kubebuilder:validation:Enum=Foo;Bar;Baz
type KubeBuilderOptionalEnum string // want "kubebuilder:validation:Optional should not be defined on a type KubeBuilderOptionalEnum"
